### PR TITLE
maxPeerHeight API enpoint for snyc progress + bugfixes

### LINKED ISSUE #1002: Actual type of BlockHeader model difficulty field in the api
The specification (openapi.yaml) says that the field type "difficulty" of the BlockHeader model is int32. In fact, the node returns a string with a number in the response of the handle `/blocks/{headerId}`

https://github.com/ergoplatform/ergo/blob/2ada64922e605bfecccddcbec91af1af52e591b9/src/main/resources/api/openapi.yaml#L793

### LINKED ISSUE #1401: Determining whether the node is still syncing and calculating sync progress
I'm trying to determining whether the node (testnet) is still syncing and if it is, calculate the current sync progress using the REST API but I can't find the datapoints in the Swagger Spec. Any pointers?

### LINKED ISSUE #1687: Int can not be casted to byte in a custom scan
For the following scan:

```
{
  "scanName": "Artworks Tracker",
  "walletInteraction": "off",
  "removeOffchain": false,
  "trackingRule": {
    "predicate": "or",
    "args" : [
      {
        "predicate": "equals",
        "register": "R7",
        "value": "0e020101"
      },
      {
        "predicate": "equals",
        "register": "R7",
        "value": "0e020102"
      },
      {
        "predicate": "equals",
        "register": "R7",
        "value": "0e020103"
      }
    ]
  }
}
```

When reaching this tx:
https://explorer.ergoplatform.com/en/transactions/8e45aa8812792834b04495be3dcf24d3edbdd2183d82aa8c716bf3659803ffa0
it fails and doesn't proceed the scan. 

Error is

```
node_1  | 00:53:32.007 INFO  [rgoref-api-dispatcher-239] o.e.n.w.ErgoWalletActor - Wallet is going to scan a block 24a0987423cf8b3824bb718a03a189aca11994f38bd7f432b10ec61ad92c4dca in the past at height 10837
node_1  | 00:53:32.029 INFO  [rgoref-api-dispatcher-239] o.e.n.w.ErgoWalletActor - Wallet is going to scan a block f58527e16abf1661d849faf4c21cc3a38cef4e0d8c2194c110839c11ecee82e6 in the past at height 10838
node_1  | 00:53:32.212 INFO  [ctor.default-dispatcher-6] s.c.n.PeerConnectionHandler - Send message MessageSpec(65: Sync) to ConnectionId(remote=/213.239.193.208:9030, local=/192.168.96.2:39650, direction=Outgoing)
node_1  | 00:53:32.212 INFO  [tor.default-dispatcher-13] s.c.n.PeerConnectionHandler - Send message MessageSpec(65: Sync) to ConnectionId(remote=/51.79.229.232:29030, local=/192.168.96.2:47536, direction=Outgoing)
node_1  | 00:53:35.168 INFO  [rgoref-api-dispatcher-239] o.e.n.w.ErgoWalletActor - Wallet actor stopped
node_1  | 00:53:35.256 ERROR [rgoref-api-dispatcher-239] o.e.n.w.ErgoWalletActor - Wallet actor restarted due to class sigmastate.SInt$ cannot be cast to class sigmastate.SByte$ (sigmastate.SInt$ and sigmastate.SByte$ are in unnamed module of loader 'app')
node_1  | java.lang.ClassCastException: class sigmastate.SInt$ cannot be cast to class sigmastate.SByte$ (sigmastate.SInt$ and sigmastate.SByte$ are in unnamed module of loader 'app')
node_1  |  at sigmastate.Values$ByteArrayConstant$.unapply(Values.scala:592)
node_1  |  at org.ergoplatform.nodeView.wallet.scanning.EqualsScanningPredicate.$anonfun$filter$2(ScanningPredicate.scala:65)
node_1  |  at org.ergoplatform.nodeView.wallet.scanning.EqualsScanningPredicate.$anonfun$filter$2$adapted(ScanningPredicate.scala:64)
node_1  |  at scala.Option.exists(Option.scala:376)
node_1  |  at org.ergoplatform.nodeView.wallet.scanning.EqualsScanningPredicate.filter(ScanningPredicate.scala:64)
node_1  |  at org.ergoplatform.nodeView.wallet.scanning.OrScanningPredicate.$anonfun$filter$9(ScanningPredicate.scala:149)
node_1  |  at org.ergoplatform.nodeView.wallet.scanning.OrScanningPredicate.$anonfun$filter$9$adapted(ScanningPredicate.scala:149)
node_1  |  at scala.collection.LinearSeqOptimized.exists(LinearSeqOptimized.scala:95)
node_1  |  at scala.collection.LinearSeqOptimized.exists$(LinearSeqOptimized.scala:92)
node_1  |  at scala.collection.immutable.List.exists(List.scala:89)
node_1  |  at org.ergoplatform.nodeView.wallet.scanning.OrScanningPredicate.filter(ScanningPredicate.scala:149)
node_1  |  at org.ergoplatform.nodeView.wallet.WalletScanLogic$.$anonfun$extractWalletOutputs$2(WalletScanLogic.scala:194)
node_1  |  at org.ergoplatform.nodeView.wallet.WalletScanLogic$.$anonfun$extractWalletOutputs$2$adapted(WalletScanLogic.scala:194)
node_1  |  at scala.collection.TraversableLike.$anonfun$filterImpl$1(TraversableLike.scala:256)
node_1  |  at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
node_1  |  at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
node_1  |  at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
node_1  |  at scala.collection.TraversableLike.filterImpl(TraversableLike.scala:255)
node_1  |  at scala.collection.TraversableLike.filterImpl$(TraversableLike.scala:249)
node_1  |  at scala.collection.AbstractTraversable.filterImpl(Traversable.scala:108)
node_1  |  at scala.collection.TraversableLike.filter(TraversableLike.scala:347)
node_1  |  at scala.collection.TraversableLike.filter$(TraversableLike.scala:347)
node_1  |  at scala.collection.AbstractTraversable.filter(Traversable.scala:108)
node_1  |  at org.ergoplatform.nodeView.wallet.WalletScanLogic$.$anonfun$extractWalletOutputs$1(WalletScanLogic.scala:194)
node_1  |  at scala.collection.TraversableLike.$anonfun$flatMap$1(TraversableLike.scala:245)
node_1  |  at scala.collection.Iterator.foreach(Iterator.scala:941)
node_1  |  at scala.collection.Iterator.foreach$(Iterator.scala:941)
node_1  |  at scala.collection.AbstractIterator.foreach(Iterator.scala:1429)
node_1  |  at scala.collection.IterableLike.foreach(Iterabl
```

### PR DESCRIPTION
**Pull request by jellymlg**
fixes #1002 
fixes #1401
fixes #1687

### CODE REVIEW

**Review by deadit — CHANGES_REQUESTED** [review]
Hey! Remove `panel` folder. It has errors. I will update the panel by myself.

File: src/main/resources/api/openapi.yaml [hunk 1]
```diff
@@ -1884,6 +1885,13 @@ components:
           minimum: 0
           example: 667
           nullable: true
+        maxPeerHeight:
+          type: integer
+          format: int32
+          description: Can be 'null' if state is empty (no peer connected since node launch)
```
Comment (pragmaxim) [inline src/main/resources/api/openapi.yaml:1891]: Explanation of peerHeight would be good here

File: src/main/scala/org/ergoplatform/local/ErgoStatsCollector.scala [hunk 2]
```diff
@@ -155,6 +164,7 @@ object ErgoStatsCollector {
                       headersScore: Option[BigInt],
                       bestFullBlockOpt: Option[ErgoFullBlock],
                       fullBlocksScore: Option[BigInt],
+                      maxPeerHeight : Option[Int],
```
Comment (pragmaxim) [inline src/main/scala/org/ergoplatform/local/ErgoStatsCollector.scala:167]: Scaladoc would be good, this is one of the fields that is not self-explanatory.

File: src/main/resources/api/openapi.yaml (hunk 1 above)
Comment (jellymlg) [reply src/main/resources/api/openapi.yaml]: added

File: src/main/scala/org/ergoplatform/local/ErgoStatsCollector.scala (hunk 2 above)
Comment (jellymlg) [reply src/main/scala/org/ergoplatform/local/ErgoStatsCollector.scala]: [69618c3](https://github.com/ergoplatform/ergo/pull/1683/commits/69618c3ecf757ba7877c95bf89b82f722774cfa7)
like this?

**Comment by kushti** [comment]
@jellymlg please resolve conflicts, and then I will test and merge. Also, added bounty.

**Comment by jellymlg** [comment]
> @jellymlg please resolve conflicts, and then I will test and merge. Also, added bounty.

done

File: src/main/scala/org/ergoplatform/modifiers/history/header/Header.scala [hunk 3]
```diff
@@ -149,7 +149,7 @@ object Header extends ApiCodecs {
       "powSolutions" -> h.powSolution.asJson,
       "nBits" -> h.nBits.asJson,
       "height" -> h.height.asJson,
-      "difficulty" -> h.requiredDifficulty.toString.asJson,
+      "difficulty" -> h.requiredDifficulty.toLong.asJson,
```
Comment (kushti) [inline src/main/scala/org/ergoplatform/modifiers/history/header/Header.scala:152]: Currently, it is hard to say which services may use this field. So better to update openapi.yaml. not the actual implementation.

**Review by kushti — APPROVED** [review]

**Pull request merged**